### PR TITLE
test: Android negative fix test

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 7cdede4</samp>

This pull request improves the `FyNumberComponent` by changing its type to `number` and adding new test cases. It also fixes a minor code style issue in the `fy-number.component.ts` file.

### <samp>🤖 Generated by Copilot at 7cdede4</samp>

> _Oh we're the coders of the `FyNumberComponent`_
> _And we test and refactor as we please_
> _We make the `value` a number and we check for negative input_
> _Heave away, heave away, heave away with ease_

### <samp>🤖 Generated by Copilot at 7cdede4</samp>

*  Simplify the type of the `innerValue`, `value`, and `onChangeCallback` properties and functions of the `FyNumberComponent` class from `string | number` to `number` ([link](https://github.com/fylein/fyle-mobile-app/pull/2338/files?diff=unified&w=0#diff-7a124d2b1279f426c898572db3f1a184b53487f32282ba714b3a1e657d374fdaL46-R46), [link](https://github.com/fylein/fyle-mobile-app/pull/2338/files?diff=unified&w=0#diff-7a124d2b1279f426c898572db3f1a184b53487f32282ba714b3a1e657d374fdaL56-R56), [link](https://github.com/fylein/fyle-mobile-app/pull/2338/files?diff=unified&w=0#diff-7a124d2b1279f426c898572db3f1a184b53487f32282ba714b3a1e657d374fdaL65-R72), [link](https://github.com/fylein/fyle-mobile-app/pull/2338/files?diff=unified&w=0#diff-7a124d2b1279f426c898572db3f1a184b53487f32282ba714b3a1e657d374fdaL87-R87))
*  Add two test cases for the `FyNumberComponent` in the `fy-number.component.spec.ts` file to check the behavior of the component when the platform is android and the `isAndroidNegativeExpensePluginEnabled` flag is enabled, and when the input is a negative number and the `isDistance` flag is true ([link](https://github.com/fylein/fyle-mobile-app/pull/2338/files?diff=unified&w=0#diff-cf5db23fa10a92ef5a35f6b3e0e0d6661e52c2c61a5e333c0757401ac3694f51R123-R147))
*  Add a comma at the end of the line in the `ngOnInit` function of the `FyNumberComponent` class to follow the code style and formatting rules ([link](https://github.com/fylein/fyle-mobile-app/pull/2338/files?diff=unified&w=0#diff-7a124d2b1279f426c898572db3f1a184b53487f32282ba714b3a1e657d374fdaL115-R115))

## Clickup
https://app.clickup.com/t/85zteyum0

## Code Coverage
Screenshot 2023-08-31 at 3 14 30 PM

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by Julias0 — APPROVED** [review]

File: src/app/shared/components/fy-number/fy-number.component.spec.ts [hunk 1]
```diff
@@ -120,6 +120,31 @@ describe('FyNumberComponent', () => {
...
+    });
+
+    it('should set this.value to null and give invalid error when given a negative number if input is distance', () => {
+      spyOn(component, 'handleChange');
+      component.isDistance = true;
+
+      component.ngOnInit();
+      fixture.detectChanges();
+      component.fc.setValue(-1);
+      expect(component.value).toBeNull();
```
Comment (Chethan-Fyle) [inline src/app/shared/components/fy-number/fy-number.component.spec.ts:145]: Where are we checking whether invalid error is thrown?

File: src/app/shared/components/fy-number/fy-number.component.spec.ts [hunk 2]
```diff
@@ -120,6 +120,31 @@ describe('FyNumberComponent', () => {
       expect(component.handleChange).not.toHaveBeenCalled();
     });
 
+    it('should enable isAndroidNegativeExpensePluginEnabled plugin when checkIfAndroidNegativeExpensePluginIsEnabled returns true', () => {
```
Comment (Chethan-Fyle) [inline src/app/shared/components/fy-number/fy-number.component.spec.ts:123]: Is the plugin name `isAndroidNegativeExpensePluginEnabled`? 🤔

**Review by Chethan-Fyle — APPROVED** [review]

**Review by Dimple16 — COMMENTED** [review]
PR is reviewed, will skip

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/shared/components/fy-number/fy-number.component.spec.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/shared/components/fy-number/fy-number.component.spec.ts:123]: Yes, it has this name only. This plugin will be true if someone faces problem in android and we enable `android-numeric-keypad`.

File: src/app/shared/components/fy-number/fy-number.component.spec.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/shared/components/fy-number/fy-number.component.spec.ts:145]: We are checking in the parent component form control that we have initialized in ngAfterViewInit

**Pull request merged**